system/libbase58/base58.c

```diff
@@ -137,9 +137,9 @@ int b58check(const void *bin, size_t binsz, const char *base58str, size_t b58sz)
         return -1;
 
     // Check number of zeros is correct AFTER verifying checksum (to avoid possibility of accessing base58str beyond the end)
-    for (i = 0; binc[i] == '\0' && base58str[i] == '1'; ++i)
+    for (i = 0; binc[i] == '\0' && base58str[i] == '1' && i < b58sz; ++i)
```

Comment:
The parameter b58sz was passed in with the string length but not used. I think it SHOULD have been used like this...

Reply:
The length test should be before the array access to prevent out of range access.

Also I think binsiz <= b58sz, so it should possibly be checking against that instead (and checking that condition)

The line below should also have the range check 1st in the condition.

Reply:
Pushed some changes.

This isn't our code, and isn't used, so it might be more prudent to disable the warning here